plugins/modules/azure_rm_publicipprefix_info.py

```diff
@@ -0,0 +1,281 @@
+            type: str
+            returned: always
+            sample: 'eastus2'
+        custom_ip_prefix:
+            description:
+                - The customIpPrefix that this prefix is associated with.
+            type: dict
+            returned: always
+            sample: {}
+'''
+try:
+    from azure.core.exceptions import ResourceNotFoundError
+except Exception:
+    # This is handled in azure_rm_common
+    pass
+
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+
+AZURE_OBJECT_CLASS = 'PublicIpPrefix'
+
+
+class AzureRMPublicIPPrefixInfo(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            name=dict(type='str'),
+            resource_group=dict(type='str'),
+            tags=dict(type='list', elements='str')
+        )
+
+        self.results = dict(
+            changed=False,
+        )
+
+        self.name = None
+        self.resource_group = None
+        self.tags = None
+
+        super(AzureRMPublicIPPrefixInfo, self).__init__(self.module_arg_spec,
+                                                        supports_check_mode=True,
+                                                        supports_tags=False,
+                                                        facts_module=True)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        result = []
+        if self.name is not None and self.resource_group is not None:
+            result = self.get_item()
+        elif self.resource_group:
+            result = self.list_resource_group()
+        else:
+            result = self.list_all()
+
+        raw = self.filter(result)
+
+        self.results['publicipprefixes'] = self.format(raw)
+
+        return self.results
+
+    def format(self, raw):
+        return [self.prefix_to_dict(item) for item in raw]
+
+    def filter(self, response):
+        return [item for item in response if self.has_tags(item.tags, self.tags)]
+
+    def prefix_to_dict(self, prefix):
+        result = dict(
+            id=prefix.id,
+            name=prefix.name,
+            tags=prefix.tags,
+            type=prefix.type,
+            location=prefix.location,
+            public_ip_address_version=prefix.public_ip_address_version,
+            prefix_length=prefix.prefix_length,
+            provisioning_state=prefix.provisioning_state,
+            etag=prefix.etag,
+            zones=prefix.zones,
+            sku=dict(),
+            ip_tags=dict(),
+            custom_ip_prefix=dict(),
+            ip_prefix=prefix.ip_prefix,
+            resource_guid=prefix.resource_guid,
+            public_ip_addresses=dict(),
+            extended_location=prefix.extended_location
+        )
+        if prefix.public_ip_addresses:
+            result['public_ip_addresses']['id'] = prefix.public_ip_addresses.id
```

Comment:
this doesn't seem to be working:
```pytb
The full traceback is:
Traceback (most recent call last):
  File "/Users/myuser/.ansible/tmp/ansible-tmp-1705314839.90448-88065-47872348921385/AnsiballZ_azure_rm_publicipprefix_info.py", line 107, in <module>
    _ansiballz_main()
  File "/Users/myuser/.ansible/tmp/ansible-tmp-1705314839.90448-88065-47872348921385/AnsiballZ_azure_rm_publicipprefix_info.py", line 99, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/Users/myuser/.ansible/tmp/ansible-tmp-1705314839.90448-88065-47872348921385/AnsiballZ_azure_rm_publicipprefix_info.py", line 47, in invoke_module
    runpy.run_module(mod_name='ansible_collections.azure.azcollection.plugins.modules.azure_rm_publicipprefix_info', init_globals=dict(_module_fqn='ansible_collections.azure.azcollection.plugins.modules.azure_rm_publicipprefix_info', _modlib_path=modlib_path),
  File "<frozen runpy>", line 226, in run_module
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/var/folders/qq/t505w6cx7fq5zq21x_94t63w0000gp/T/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload_8fgnglfo/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_publicipprefix_info.py", line 281, in <module>
  File "/var/folders/qq/t505w6cx7fq5zq21x_94t63w0000gp/T/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload_8fgnglfo/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_publicipprefix_info.py", line 277, in main
  File "/var/folders/qq/t505w6cx7fq5zq21x_94t63w0000gp/T/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload_8fgnglfo/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_publicipprefix_info.py", line 190, in __init__
  File "/var/folders/qq/t505w6cx7fq5zq21x_94t63w0000gp/T/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload_8fgnglfo/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py", line 446, in __init__
  File "/var/folders/qq/t505w6cx7fq5zq21x_94t63w0000gp/T/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload_8fgnglfo/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_publicipprefix_info.py", line 209, in exec_module
  File "/var/folders/qq/t505w6cx7fq5zq21x_94t63w0000gp/T/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload_8fgnglfo/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_publicipprefix_info.py", line 214, in format
  File "/var/folders/qq/t505w6cx7fq5zq21x_94t63w0000gp/T/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload_8fgnglfo/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_publicipprefix_info.py", line 214, in <listcomp>
  File "/var/folders/qq/t505w6cx7fq5zq21x_94t63w0000gp/T/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload_8fgnglfo/ansible_azure.azcollection.azure_rm_publicipprefix_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_publicipprefix_info.py", line 240, in prefix_to_dict
AttributeError: 'list' object has no attribute 'id'
```

`prefix.public_ip_addresses` is a list, so this should probably be

```suggestion
            public_ip_addresses=list(),
            extended_location=prefix.extended_location
        )
        if prefix.public_ip_addresses:
            result['public_ip_addresses'] = [x.id for x in prefix.public_ip_addresses]
```